cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -313,6 +322,22 @@ def supplemental_index_information(cls, modulestore, structure):
         """
         CourseAboutSearchIndexer.index_about_information(modulestore, structure)
 
+    @classmethod
+    def supplemental_fields(cls, item):
+        """
+        Add location path to the item object
+        """
+        location_path = []
+        parent = item
+        while parent is not None:
+            location_path.append(parent.display_name_with_default)
+            parent = parent.get_parent()
+        location_path.reverse()
+        return {
+            "course_name": location_path[0],
+            "location": location_path[1:4]
```

Comment:
NIT: Perhaps a comment is in order here, to explain why we're taking those 3 elements from `location_path` to form a location.